src/metabase/api/dataset.clj

```diff
@@ -56,24 +59,35 @@
     (when (int? table-id)
       (events/publish-event! :table-read (assoc (db/select-one Table :id table-id) :actor_id api/*current-user-id*))))
   ;; add sensible constraints for results limits on our query
-  (let [source-card-id (query->source-card-id query)
-        source-card    (when source-card-id
-                         (db/select-one [Card :result_metadata :dataset] :id source-card-id))
-        info           (cond-> {:executed-by api/*current-user-id*
-                                :context     context
-                                :card-id     source-card-id}
-                         (:dataset source-card)
-                         (assoc :metadata/dataset-metadata (:result_metadata source-card)))]
+  (let [source-card-id   (query->source-card-id query)
+        source-card      (when source-card-id
+                           (db/select-one [Card :result_metadata :dataset] :id source-card-id))
+        dataset-metadata (if (and (:dataset source-card) dataset-metadata)
```

Comment:
If the source card is a dataset and dataset metadata is present, we'll combine them before passing it into `:info`.